File: app-ui/src/models/demoData/Album.ts
import { NamedEntityWithImage } from "./NamedEntityWithImage";
import {Artist} from "./Artist";
import {Track} from "./Track";
import {LazyShim} from "../LazyShim";

export class Album extends NamedEntityWithImage {
    public artistId: number = 0;
    public releaseDate: Date = new Date();
    public artist: LazyShim<Artist> = new LazyShim<Artist>();
    public tracks: LazyShim<Track[]> = new LazyShim<Track[]>();
}

File: app-api/LibElectronAppDemo/Database/Models/Album.cs
using System.ComponentModel.DataAnnotations.Schema;
using LibSqlite3Orm.Abstract.Orm;
using Newtonsoft.Json;

namespace LibElectronAppDemo.Database.Models;

public class Album : NamedEntityWithImage
{
    public long ArtistId { get; set; }
    public DateOnly ReleaseDate { get; set; }
    [NotMapped]
    public Lazy<Artist> Artist { get; set; }
    [JsonIgnore]
    [NotMapped]
    public Lazy<ISqliteQueryable<Track>> Tracks { get; set; }
}